### ISSUE TITLE
Request for Comments: make regolith theme as global as possible

### ISSUE THREAD

**Issue by ploum** [regolith-linux/regolith-desktop#165]
Currently, in Regolith, if you want to switch from a theme to another, you have two lines to change : one for the color panel, the other for the gtk-theme (which includes itself the icon theme).

The idea here is to make it even easier with one line: loading the colours would be done in the style/theme file instead of the Xressources.

To go even further, those theme could be applied to common applications automatically (vim, mutt and zsh comes to my mind but there might be others low hanging fruits).

Benefits:
- An unified Regolith experience.
- Easy to switch theme for your whole system.

Cons:
- Composite configurations would have to make their own theme.

I think it all boils down to a philosophical question: is it the goal of Regolith to makes it easier to switch between good looking well tested theme at the price of making it a bit harder to have a fine grained configuration of your system?

**Comment by kgilmer** [comment]
A core design goal is to make a tiling desktop environment that's approachable and fun for those that otherwise wouldn't have the time or skills to build their own.  So, making the look and feel easily changeable is definitely in scope for the project.  The trick is to enable this *without* regression on hand-tuned UI changes.  Meaning, if (not suggesting this was proposed) some theme engine generates a binary blob that is then used by native system apps to read theme info, this would be bad for those that prefer to modify configuration files.  Another regression would be to require specific  (machine-oriented) changes to be applied exactly, across a large set of files.

I've read about some projects that attempt to do this at the application layer.  Base16 comes to mind as does pywall and oomix, although I'm only just passingly familiar with these projects.  Ideally we could stand taller on some of these shoulders...

**Comment by brandondees** [comment]
Sounds nice to me

**Comment by kgilmer** [comment]
I believe the existing Xresources/GNOME theme logic in Regolith can be enhanced to support this use case in a natural way.  To do this the existing `regolith-styles` package will be decomposed into a root set of definitions (eg `/etc/regolith/styles/i3-wm`) which only define the structure of what can be specified, and then a set of mutually exclusive packages that all provide the virtual package `regolith-theme`.  All of these would provide the `/etc/regolith/styles/root` file which is loaded upon X session start.  Debian package scripts such as `postinst` can be utilized to perform setting such as configuring GTK themes and restarting i3wm as needed.  This should allow themes to be hot-swapped, and easily configurable via `apt` or a package front-end such as `synaptic`.

In summary, `regolith-styles` becomes:
`regolith-styles` (only defines the Xresource keys)
`regolith-theme` (a virtual package declaration)
`regolith-theme-default`, `regolith-theme-solarized-dark`, etc

Users wishing to do their own thing can continue to stage their own Xresource file in `~/.Xresources-regolith` and whatever is in `/etc/regolith/styles` will be ignored unless explicitly referenced.

**Comment by kgilmer** [comment]
I will move forward with implementation.  Anyone with concerns is welcome to chime in...

**Comment by kgilmer** [comment]
The design feels good; the change also allows for all thematic dependencies to be pushed down to the specific packages that require them rather than monolithicly being associated with the top-level `regolith-desktop` package.

**Comment by kgilmer** [comment]
Will let canonical's package builder chew on these packages over night.  What's left:

1. identify gtk themes for solarized-light and gruvbox
2. add postinst scripts to have the gtk, icon, and Xresources contents applied upon install
3. test and iterate

**Comment by kgilmer** [comment]
This stuff is pretty raw so I'll be staging packages in `regolith-experimental` rather than `regolith-unstable` until things look ok.

**Comment by kgilmer** [comment]
Packages are staged for testing: https://launchpad.net/~kgilmer/+archive/ubuntu/regolith-experimental/+sourcepub/10796483/+listing-archive-extra

**Comment by arendina** [comment]
I noticed that if you choose a light theme, then you have the bar that is following the color rules of the light theme. It would be nice to have the chance to load a theme for the bar and one for the rest of the desktop. So you can mix a dark bar with a light theme.

**Comment by kgilmer** [comment]
I see what you're saying @arendina .  The current scheme doesn't support that level of granularity.  It's easy enough to change the [bar colors](https://github.com/regolith-linux/regolith-styles/blob/R1.2/Xresources/i3-wm#L8-L10) manually, but at the package level it would have to be that there is one package with a dark bar and perhaps another package with a light bar.  I'm not aware of a solution that would allow for this fine grained control but still provide a simple way for new users to configure the look.

**Comment by kgilmer** [comment]
In testing, was able to modify the default look to this with one `apt` command and restart my X session: 

![image](https://user-images.githubusercontent.com/49683/70729985-0e242a00-1cb9-11ea-9fab-8744760c9b21.png)

**Comment by kgilmer** [comment]
Closing this as looks are now packaged as discrete entities that include all theme data.
